module-2/backend/week_one.md

```diff
@@ -5,30 +5,79 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+
 6. What types of variables are accessible in our view templates without explicitly passing them?
+params
+
 7. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
+  @count = 1 
   
   ```ruby
-  get '/horses' do
+  get '/horses' do  
     erb :index
   end
   ```
 
 8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?
+name = 'Mr. Ed'
+
 9. What's the purpose of ERB?
+ERB allows us to run Ruby code in HTML - we can make HTML dynamic. 
+
 10. Why do I need a development AND test database?
+I don't know. My guess is that the test database will often be deleted and recreated, can be used as a kind of sketch board. Whereas the development environment is a place to build something much closer to the production environment (with imported data, etc).
+
 11. What's responsive design?
+Responsive design allows the web page to respond to screen size.
+
 12. What is CRUD and why is it important?
+Create, Read, Update, Destroy - these are the main functionality for all web applications. You can create, read, update, or destroy any given thing in an application. 
+
 13. What does HTTP stand for? 
+Hyper Text Transfer Protocol
+
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+<%= %> - this one will be viewed in the HTML displayed on the website
+<% %> - this one will not be viewed in the HTML
 15. What's an ORM?
+Object relational mapper - it is a wrapper written in an object oriented language like Ruby or Python that allows for interaction with a database (such as SQL). In this case, each row (attribute) in a database corresponds to a object. Those rows can be manipulated and used just like regular objects. From a design perspective, this is especially beneficial because it pushes all state management to the database. The ORM simply writes SQL in such a way that we can interact with the database using objects.  
 16. What's the most commonly used ORM in ruby (Sinatra & Rails)?
+ActiveRecord
 17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+/restaurants - GET - shows all restaurants - READ
+/restaurants/:id - GET - views specific restaurant - READ
+/restaurants/new - GET - allows viewing of form to create new restaurant - CREATE
+/restaurants/new - POST - allows creation through form submission of new restaurant - CREATE
+/restaurants/edit - GET - allows viewing of existing restaurants, can click to edit - UPDATE
+/restaurants/edit/:id - POST - allows editing of a specific restaurant - UPDATE
+/restaurants/edit/:id - DELETE - allows deleting of specific object - DELETE

```

Comment:
For these last three:

/restaurants/:id/edit - GET - returns a form for editing a restaurant with an id of `:id`
/restaurants/:id      - PUT - updates the restaurant with an id of `:id` in the database
/restaurants/:id      - DELETE - deletes the restaurant with an id of `:id` in the database